xbmc/windowing/X11/GLContextEGL.cpp

```diff
@@ -17,17 +17,22 @@
  *  <http://www.gnu.org/licenses/>.
  *
  */
+#include "system_gl.h"
```

Comment:
Indeed it is a bit crap, but hides the complexity of the ugly, commonly used, conditional includes in many files. Nevertheless I'm going to follow your advice.